src/go/cmd/http-relay-client/client/client.go

```diff
@@ -564,6 +576,7 @@ func (c *Client) handleRequest(remote *http.Client, local *http.Client, pbreq *p
 	defer span.End()
 
 	resp, hresp, err := makeBackendRequest(ctx, local, req, id)
+	defer hresp.Body.Close()
```

Comment:
Why are we closing the Body here? Does the comment below not apply anymore?

// hresp.Body is either closed from streamToBackend() or streamBytes()

Also, we are not trying segfaulting if err != nil since then Body is nil and we cannot close it.

Reply:
+1 I'd revert that line and see to have missed the change